packages/ts-client/src/ConnectPromise.ts

```diff
@@ -0,0 +1,40 @@
+import { Deferred } from './webrtc/deferred';
+import type { FishjamClient } from './FishjamClient';
+
+export class ConnectPromise<P, T> {
```

Comment:
WDYT about using just promise instead of full new class?
```ts
export function connectPromiseFun<T, P>(fishjamClient: FishjamClient<P, T>) {
  const promise = new Promise<void>((resolve, reject) => {
    const onSuccess = () => {
      clearCallbacks();
      resolve();
    };
    const onError = () => {
      clearCallbacks();
      reject();
    };

    fishjamClient.on('joined', onSuccess);
    fishjamClient.on('joinError', onError);
    fishjamClient.on('authError', onError);
    fishjamClient.on('socketError', onError);

    const clearCallbacks = () => {
      fishjamClient.removeListener('joined', onSuccess);
      fishjamClient.removeListener('joinError', onError);
      fishjamClient.removeListener('authError', onError);
      fishjamClient.removeListener('socketError', onError);
    };
  });

  return promise;
}
```

> [!NOTE]
> Name `connectPromiseFun` is just example, could be something more descriptive - like `connectEventsHandler`?

Reply:
It's nice and elegant. I like it!